### ISSUE TITLE
Fatal errors on both Lite and Pro versions after updating to the latest 3.1.2 version - new product, using the [downloads] shortcode

### ISSUE THREAD

**Issue by napoleaowingert** [awesomemotive/easy-digital-downloads#9652]
### Expected behavior
We have a page where we are using the ```[downloads]``` shortcode and everything works well.
So if we create a new "Test Product", and flag it as a variable pricing product, we expected that the page still works.

### Actual behavior
When we create a new product and flag it as variable pricing without filling in more data, the ```[downloads]``` shortcode throws fatal errors - so the mentioned page does not work anymore.

### Steps to reproduce the behavior
1) Create a page with a title and the [downloads] shortcode only. It is important to create this page first, otherwise, you can't even publish the page later on - it fails due to the mentioned fatal error;
2) Create a new download with a title only and flag it as variable pricing by checking the "Enable variable pricing" checkbox;
3) That's it. Publish the product;
4) Now open the page where you have the [downloads] shortcode: there is a fatal error.

### Information (if a specific version is affected):
PHP Version: 8.0.28

EDD Version (or branch): 3.1.2

WordPress Version: 6.2.2

Any other relevant information:

Tested with Lite and Pro versions, same results.

Active Theme: Twenty Twenty-Three
Active Plugins: EDD core lite/pro only.

It looks like the variable prices meta key/value pair is not stored but when retrieving prices it returns the following:
```
Array
(
    [0] => 
)
```
then the ```easy-digital-downloads-pro/includes/template-functions.php``` file tries to loop through the prices but there is no data.

Here is the fatal error:
```
PHP Fatal error:  Uncaught TypeError: Cannot access offset of type string on string in /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php:306
Stack trace:
#0 /www/private_server/public/wp-includes/class-wp-hook.php(308): edd_purchase_variable_pricing(422367, Array)
#1 /www/private_server/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#2 /www/private_server/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(184): do_action('edd_purchase_li...', 422367, Array)
#4 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/templates/shortcode-content-cart-button.php(2): edd_get_purchase_link(Array)
#5 /www/private_server/public/wp-includes/template.php(785): require('/www/private_server...')
#6 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(719): load_template('/www/private_server...', false)
#7 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(652): edd_locate_template(Array, true, false)
#8 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/templates/shortcode-download.php(46): edd_get_template_part('shortcode', 'content-cart-bu...')
#9 /www/private_server/public/wp-includes/template.php(785): require('/www/private_server...')
#10 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(719): load_template('/www/private_server...', false)
#11 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(652): edd_locate_template(Array, true, false)
#12 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php(1010): edd_get_template_part('shortcode', 'download')
#13 /www/private_server/public/wp-includes/class-wp-hook.php(308): edd_download_shortcode_item(Array, 1)
#14 /www/private_server/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#15 /www/private_server/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#16 /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/shortcodes.php(602): do_action('edd_download_sh...', Array, 1)
#17 /www/private_server/public/wp-includes/shortcodes.php(355): edd_downloads_query(Array, '', 'downloads')
#18 [internal function]: do_shortcode_tag(Array)
#19 /www/private_server/public/wp-includes/shortcodes.php(227): preg_replace_callback('/\\[(\\[?)(downlo...', 'do_shortcode_ta...', '[downloads]\n')
#20 /www/private_server/public/wp-includes/class-wp-hook.php(308): do_shortcode('[downloads]\n')
#21 /www/private_server/public/wp-includes/plugin.php(205): WP_Hook->apply_filters('[downloads]\n', Array)
#22 /www/private_server/public/wp-includes/blocks/post-content.php(54): apply_filters('the_content', '<!-- wp:shortco...')
#23 /www/private_server/public/wp-includes/class-wp-block.php(258): render_block_core_post_content(Array, '<!-- wp:shortco...', Object(WP_Block))
#24 /www/private_server/public/wp-includes/class-wp-block.php(244): WP_Block->render()
#25 /www/private_server/public/wp-includes/blocks.php(1072): WP_Block->render()
#26 /www/private_server/public/wp-includes/blocks.php(1110): render_block(Array)
#27 /www/private_server/public/wp-includes/block-template.php(242): do_blocks('<!-- wp:templat...')
#28 /www/private_server/public/wp-includes/template-canvas.php(12): get_the_block_template_html()
#29 /www/private_server/public/wp-includes/template-loader.php(106): include('/www/private_server...')
#30 /www/private_server/public/wp-blog-header.php(19): require_once('/www/private_server...')
#31 /www/private_server/public/index.php(17): require('/www/private_server...')
#32 {main}
  thrown in /www/private_server/public/wp-content/plugins/easy-digital-downloads-pro/includes/template-functions.php on line 306
```

If you could check please if that happens on your end as well, thank you.

**Comment by robincornett** [comment]
@napoleaowingert thank you for the report--I can replicate the issue. I'm going to copy this to our working repository.

**Comment by efmcyrill** [comment]
Hi @robincornett, will any fix be released in the next few days? We are a bit stuck waiting for this to be fixed before we can update EDD plugin.

**Comment by robincornett** [comment]
We are actively working to finish the next release, which will include a fix for this, yes. If you want to apply the main part of fix we're planning to your copy of 3.1.2, you can change [this line](https://github.com/awesomemotive/easy-digital-downloads/blob/main/includes/class-edd-download.php#L309) to this:

```php
return array_filter( (array) apply_filters( 'edd_get_variable_prices', $this->prices, $this->ID ) );
```

I confess that I'm curious--what is the benefit of flagging a product as variable pricing, but not adding any prices? I confess that I would have expected this scenario to be an edge case.

Thanks again for the report and your patience.

**Comment by napoleaowingert** [comment]
Hi @robincornett.

We simply updated EDD on a stage server and suddently we had a fatal error - we think that shouldn't happen.

The reason was probably some test product that was saved like that by mistake while testing something else, in our case.

Now eg. if someone is creating a new product on a production server and for any reason has to stop for a moment and quickly saves the product like that, then all pages where the shortcode is in use might break, etc.

**Comment by cklosowski** [comment]
We've fixed this in the EDD 3.1.3 release.
